src/OpenLoco/Windows/TerraForm.cpp

```diff
@@ -648,15 +622,33 @@ namespace OpenLoco::Ui::Windows::Terraform
                 // Note: this is not the same as the randomDirection above as it is the trees rotation
                 args.rotation = rng.randNext(3);
                 args.colour = 0;
-                auto type = getRandomTreeTypeFromSurface(newLoc, false);
-                if (type)
+                auto type = getTreeType(newLoc, false);
+                if (!type)
+                {
+                    continue;
+                }
+                args.type = *type;
+                args.buildImmediately = true;
+                args.requiresFullClearance = true;
+
+                // First query if we can place a tree at this location; skip if we can't.
+                auto queryRes = do_23(0, args);
+                if (queryRes == GameCommands::FAILURE)
                 {
-                    args.type = *type;
-                    args.buildImmediately = true;
-                    args.requiresFullClearance = true;
-                    do_23(GameCommands::Flags::apply, args);
+                    numErrors++;
+                    continue;
                 }
+
+                // Actually place the tree
+                do_23(GameCommands::Flags::apply, args);
             }
+
+            // Have we placed any trees?
+            if (numErrors < numPlacements)
```

Comment:
why not just `numErrors != 0`

Reply:
That would prevent the error below from showing up entirely. (Where numErrors == numPlacements, numErrors != 0 holds as well)

Reply:
oh yeah my bad